pkg/migration/azure_crs.go

```diff
@@ -0,0 +1,392 @@
+package migration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+
+	provider "github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	cabpkv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	kubeadm "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	EncryptionSecret = "EncryptionSecret"
+)
+
+func (m *azureMigrator) createEncryptionConfigSecret(ctx context.Context) error {
+	var origEncryptionSecret *corev1.Secret
+	{
+		obj, exists := m.crs[EncryptionSecret]
+		if !exists {
+			return microerror.Mask(fmt.Errorf("encryption secret not found"))
+		}
+
+		origEncryptionSecret, ok := obj.(*corev1.Secret)
+		if !ok {
+			return microerror.Mask(fmt.Errorf("can't convert obj (%T) to %T", obj, origEncryptionSecret))
+		}
+	}
+
+	encryptionConfigTmpl := `
```

Comment:
WDYT, should I put these templates inside templates also into a file under `templates/` and load from there or is it better to have the embedded content inlined here?

Reply:
I personally prefer having template files because it's easier to spot syntax errors with the IDE.
On the other hand this is a limited-time operator and I can leave with inline templates so I'll leave the decision to you.

Reply:
I'll leave it as-is for now. I do agree with template files but since we seem to be in rush, I don't feel like polishing...